File: Labs/Lab 5/CS/Lab5/Strings.cs
using System;

namespace PolymorphismDemo
{
    public class Strings
    {
        protected string value;

        public Strings(string value)
        {
            this.value = value;
        }

        public virtual int GetLength()
        {
            return value.Length;
        }

        public virtual int CountSymbol(char symbol)
        {
            int count = 0;
            foreach (var ch in value)
            {
                if (ch == symbol)
                {
                    count++;
                }
            }
            return count;
        }
    }
}

File: Labs/Lab 5/TS/Strings.ts
export class Strings {
  protected value: string;

  constructor(value: string) {
    this.value = value;
  }

  getLength(): number {
    return this.value.length;
  }

  countSymbol(symbol: string): number {
    let count = 0;
    for (let c of this.value) {
      if (c === symbol) {
        count++;
      }
    }
    return count;
  }
}
